### PR TITLE
Re-enable the build of OpenCL unit tests

### PR DESCRIPTION
**Pull request by yannlabou**
Trying to locate OpenCL under CUDA install directory.

### CODE REVIEW

**Comment by ihhub** [comment]
@yannlabou here are no changes in the code. You might missed something :)

**Comment by ihhub** [comment]
I'm suggesting to print all files with absolute path so we would know which files we have there.

**Comment by ihhub** [comment]
Shall we try `dir` command in test script? You could have a look 
[here](https://stackoverflow.com/questions/15214486/command-to-list-all-files-in-a-folder-as-well-as-sub-folders-in-windows)

**Comment by yannlabou** [comment]
> Shall we try `dir` command in test script? You could have a look
> [here](https://stackoverflow.com/questions/15214486/command-to-list-all-files-in-a-folder-as-well-as-sub-folders-in-windows)

Sure.

**Comment by ihhub** [comment]
`C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v8.0\lib\Win32\OpenCL.lib` for x86 build and `C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v8.0\lib\x64\OpenCL.lib` for x64. Is this what we need?

**Comment by yannlabou** [comment]
> `C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v8.0\lib\Win32\OpenCL.lib` for x86 build and `C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v8.0\lib\x64\OpenCL.lib` for x64. Is this what we need?

and `C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v8.0\include\CL\` :)

**Comment by yannlabou** [comment]
Next time I'll go for the simplest solution. I promise.

**Comment by ihhub** [comment]
Let's try. I have a feeling that we need to copy *.dll* files as well. Visual Studio on Windows does this for CUDA projects.

**Comment by yannlabou** [comment]
> Let's try. I have a feeling that we need to copy _.dll_ files as well. Visual Studio on Windows does this for CUDA projects.

I downloaded the cuda exe this morning, extracted it as I wasn't able to install and I tested on my Windows machine. I was able to run the tests just by linking to the lib. Let's see.

**Comment by ihhub** [comment]
There are coupld of them like cudart64_80.dll Which one are you looking for?

**Comment by yannlabou** [comment]
> There are couple of them like cudart64_80.dll Which one are you looking for?

OK, that's fine. I expected opencl.dll. I will give the CUDA DLLs a try.

**Comment by ihhub** [comment]
Looks like the only way is to do not run OpenCL unit tests executable which is fine after all. I'll review the changes shortly.

File: appveyor.yml [hunk 1]
```diff
@@ -24,15 +24,16 @@ clone_folder: C:\projects\penguinv
 # scripts to run before build
 before_build:
   - cmd: call script\appveyor\cuda_install.cmd
+  - cmd: call script\appveyor\find_cuda_opencl.cmd
```
[the remark this answers is not in the document]
Comment (yannlabou) [reply appveyor.yml]: Yes.

File: script/appveyor/find_cuda_opencl.cmd [hunk 2]
```diff
@@ -0,0 +1,11 @@
+@echo off
```
[the remark this answers is not in the document]
Comment (yannlabou) [reply script/appveyor/find_cuda_opencl.cmd]: Yes.

File: test/unit_tests/CMakeLists.txt [hunk 3]
```diff
@@ -22,9 +22,7 @@ find_package(Threads REQUIRED)
 option(PENGUINV_BUILD_UNIT_TEST_OPENCL "Build OpenCL unit tests" ON)
 if(${PENGUINV_BUILD_UNIT_TEST_OPENCL})
     find_package(OpenCL QUIET)
-    if (OpenCL_FOUND)
```
[the remark this answers is not in the document]
Comment (yannlabou) [reply test/unit_tests/CMakeLists.txt]: Still it's valid. The user should then call `cmake -DPENGUINV_BUILD_UNIT_TEST_OPENCL=OFF`.

File: appveyor.yml [hunk 4]
```diff
@@ -24,15 +24,16 @@ clone_folder: C:\projects\penguinv
...
 
 build_script:
   - cmd: cd C:\projects\penguinv
   - cmd: md build
   - cmd: cd build
-  - cmd: if "%platform%"=="x64" cmake -G "Visual Studio 14 2015 Win64" -DPENGUINV_BUILD_UNIT_TEST_OPENCL=OFF ..
-  - cmd: if "%platform%"=="x86" cmake -G "Visual Studio 14 2015" -DPENGUINV_BUILD_UNIT_TEST_OPENCL=OFF ..
+  - cmd: if "%platform%"=="x64" cmake -G "Visual Studio 14 2015 Win64" -DPENGUINV_BUILD_UNIT_TEST_OPENCL=ON ..
```
[the remark this answers is not in the document]
Comment (yannlabou) [reply appveyor.yml]: Sure.

File: appveyor.yml [hunk 5]
```diff
@@ -24,15 +24,16 @@ clone_folder: C:\projects\penguinv
...
+  - cmd: if "%platform%"=="x86" cmake -G "Visual Studio 14 2015" -DPENGUINV_BUILD_UNIT_TEST_OPENCL=ON ..
   - cmd: cmake --build . --config %configuration%
 
 test_script:
-  - cmd: ctest -C %configuration% --extra-verbose
+  - cmd: ctest -C %configuration% -E unit_tests_opencl --extra-verbose
```
[the remark this answers is not in the document]
Comment (yannlabou) [reply appveyor.yml]: Yes. Will do.

File: test/unit_tests/opencl/CMakeLists.txt [hunk 6]
```diff
@@ -36,10 +36,10 @@ endif()
 # Move the the executable file created by add_executable to the <build_directory>/bin subfolder
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/bin)
 add_executable(${TARGET_NAME} ${SRCS})
-target_link_libraries(${TARGET_NAME}
```
[the remark this answers is not in the document]
Comment (yannlabou) [reply test/unit_tests/opencl/CMakeLists.txt]: Yes it is. I just aligned the indentation.

File: test/unit_tests/CMakeLists.txt [hunk 7]
```diff
@@ -22,9 +22,7 @@ find_package(Threads REQUIRED)
 option(PENGUINV_BUILD_UNIT_TEST_OPENCL "Build OpenCL unit tests" ON)
 if(${PENGUINV_BUILD_UNIT_TEST_OPENCL})
```
[the remark this answers is not in the document]
Comment (yannlabou) [reply test/unit_tests/CMakeLists.txt]: I think it's better to let the user decide whether or not he wants to build the tests. Our default is ON.

**Comment by ihhub** [comment]
@yannlabou , thanks a lot again for your continuous support of the project!

**Pull request merged**